File: Service/Z2data.Core/Domain/Events/Event.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Z2data.Core.Domain.Events
{
   public class Event:AuditEntity
    {
        public string Code { get; set; }
        public Place Place { get; set; }
        public EventDetails Details { get; set; }
        public EventFeature Features { get; set; }
        public EffectRange EffectRange { get; set; }
        public AssociatedEvent AssociatedEvent { get; set; }
        public IList<AffectedLocation> AffectedLocations { get; set; }
        public bool IsAlert { get; set; }
        public IList<Event> Updates { get; set; }
    }
}

File: Front-end/src/app/events/meta-models/event.ts
import { AuditEntity } from '../../shared/auditentity';
import { Place } from '../../events/meta-models/Place';
import { EventDetails } from '../../events/meta-models/event-details';
import { EventFeature } from '../../events/meta-models/event-feature';
import { EffectRange } from '../../events/meta-models/effect-range';
import { AssociatedEvent } from '../../events/meta-models/associated-event';
import { AffectedLocation } from '../../events/meta-models/affected-location';
export interface Event extends AuditEntity {
		code: string;
		place: Place;
		details: EventDetails;
		features: EventFeature;
		effectRange: EffectRange;
		associatedEvent: AssociatedEvent;
		affectedLocations: AffectedLocation[];
		isAlert: boolean;
		updates: any[];
}
